jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/TaskCountLoadCalculator.java

```diff
@@ -0,0 +1,96 @@
+
+public class TaskCountLoadCalculator extends LoadCalculatorImpl {
+	private static final String IDENTIFIER = "TaskCountLoadCalculator";
+	private static final Logger logger = LoggerFactory.getLogger(TaskCountLoadCalculator.class);
+    private static final String SINGLE_USER_QUERY = 
+            "select new Integer(count(t)) "
+            + "from TaskImpl t "
+            + "where t.taskData.actualOwner.id = :owner and t.taskData.status in ('Reserved', 'InProgress', 'Suspended') ";
+    private static final String MULTI_USER_QUERY = 
+            "select new org.jbpm.services.task.assignment.impl.AssignmentImpl(t.taskData.actualOwner.id, count(t)) "
+            + "from TaskImpl t "
+            + "where t.taskData.actualOwner.id in (:owners) and t.taskData.status in ('Reserved', 'InProgress', 'Suspended') "
+            + "group by t.taskData.actualOwner "
+            + "order by count(t) asc, t.taskData.actualOwner.id asc";
+    
+    
+    public TaskCountLoadCalculator() {
+    	super(IDENTIFIER);
+    }
+
+    Function<AssignmentImpl, String> assignKey = (assignment) -> { return assignment.getUser(); };
+    
+    @Override
+	public UserTaskLoad getUserTaskLoad(User user, TaskContext context) {
+		UserTaskLoad load = new UserTaskLoad(getIdentifier(), user);
+        TaskPersistenceContext persistenceContext = ((org.jbpm.services.task.commands.TaskContext)context).getPersistenceContext();
+        Map<String, Object> params = new HashMap<>();
+        params.put("owner", user.getId());
+        
+        logger.debug("DB query to be used for finding assignments :: '{}'", getSingleUserQuery());
+        List<Integer> assignmentCounts = persistenceContext.queryStringWithParametersInTransaction(getSingleUserQuery(), params, ClassUtil.<List<Integer>>castClass(List.class));
+        if (assignmentCounts != null && !assignmentCounts.isEmpty()) {
+        	load.setCalculatedLoad(new Double(assignmentCounts.get(0)));
+        } else {
+        	load.setCalculatedLoad(new Double(0));
+        }
+		return load;
+	}
+    
+
+	@Override
+	public Collection<UserTaskLoad> getUserTaskLoads(List<User> users, TaskContext context) {
+		Collection<UserTaskLoad> userTaskLoads = new ArrayList<>();
+		List<String> userIds = users.stream().map(user -> {return user.getId();}).collect(Collectors.toList());
+		
+        TaskPersistenceContext persistenceContext = ((org.jbpm.services.task.commands.TaskContext)context).getPersistenceContext();
+        Map<String, Object> params = new HashMap<>();
+        params.put("owners", userIds);
+        
+        logger.debug("DB query to be used for finding assignments :: '{}'", getMultiUserQuery());
+        List<AssignmentImpl> assignments = persistenceContext.queryStringWithParametersInTransaction(getMultiUserQuery(), params, ClassUtil.<List<AssignmentImpl>>castClass(List.class));
+        Map<String,AssignmentImpl> assignmentMap = assignments.stream().collect(Collectors.toMap(assignKey,(assign)->assign));
```

Comment:
The assignKey references the earlier defined Function<AssignmentImpl,String> assignKey...
And yes, it could be replaced with a method, but why would a method be preferable to using the Function?
I would be more inclined to change from using the external Function to using an inline function.